tests/unit/test_incremental.py

```diff
@@ -65,6 +65,20 @@ def test_sync_table(self, mocked_register_hstore):
         self.assertEqual(expected_state_replication_key_value,
                          actual_state['bookmarks'][self.stream['tap_stream_id']]['replication_key_value'],
                          )
+    
+    @patch("psycopg2.extras.register_hstore")
+    def test_sync_table_always_return_1_row(self, mocked_register_hstore):
+        with patch('psycopg2.connect') as mocked_connect:
+            mocked_connect.return_value.__enter__.return_value = MockedConnectWithZeroRowAtFirst()
+            desired_columns = ['foo_key']
+            self.state['bookmarks'] = {}
+            expected_state_replication_key_value = MockedConnect.cursor.return_value
+            actual_state = incremental.sync_table(self.conn_config, self.stream, self.state, desired_columns, self.md_map)
+            mocked_register_hstore.assert_called()
+
+            self.assertEqual(expected_state_replication_key_value,
+                            actual_state['bookmarks'][self.stream['tap_stream_id']]['replication_key_value'],
+                            )
```

Comment:
```suggestion
    @patch("psycopg2.extras.register_hstore")
    def test_sync_table_always_return_1_row(self, mocked_register_hstore):
        """Test for sync_table if it returns 1 row"""
        desired_columns = ['foo_key']
        self.state['bookmarks'] = {}
        MockedConnect.cursor.counter_limit = 0
        expected_state_replication_key_value = MockedConnect.cursor.fetchone_return_value[0]

        actual_state = incremental.sync_table(self.conn_config, self.stream, self.state, desired_columns, self.md_map)
        mocked_register_hstore.assert_called()

        self.assertEqual(expected_state_replication_key_value,
                         actual_state['bookmarks'][self.stream['tap_stream_id']]['replication_key_value'],
                         )
```

👏 Impressive, it's really smart of you that you have created a dedicated `MockedConnectWithZeroRowAtFirst` to patch the behavior of how cursor iterating the results set. After playing around, I've figured out that we can achieve the same by doing 2 simple steps in this particular unit-test as follows
- Set  `counter_limit = 0` to stop iterating
- The mock `cursor.fetchone_return_value` always return `5`, we can use that to assert the unit-test

By doing that way, you can re-use the same object with slightly modification 👍. Don't hesitate to let me know what are your thoughts on this approach!

Reply:
Thanks, sir 🙇